tf2onnx/graph.py

```diff
@@ -1169,6 +1181,20 @@ def delete_unused_nodes(self, outputs_name):
                     body_graph.delete_unused_nodes(body_graph.outputs)
         self.reset_nodes(related_nodes)
 
+    def delete_nodes_without_dependency(self, to_delete):
+        """Delete nodes in `to_delete` without third-party dependency."""
+        for n in to_delete:
+            can_delete = True
```

Comment:
optional:
find all consumers into a set, and use issubset, so the for ... for.... is not need.